### ISSUE TITLE
core: `diff` subcommand silently ignores non-existing flags

### ISSUE THREAD

**Issue by l0b0** [aws/aws-cdk#27244]
### Describe the bug

Running `cdk diff` with an unknown flag does not result in any kind of indication that there is a problem.

### Expected Behavior

`cdk diff [other options] --no-such-flag` should:

- Indicate clearly that the `--no-such-flag` is unknown. This way users immediately recognise where their mistakes are, rather than having to correlate each flag with the documentation to detect the error. Ideally, if there are multiple unknown flags, each one of them should be printed in a similar error message.
- Exit with a _unique non-zero exit code,_ rather than exit code 0/1. This way anyone automating CDK can tell the difference between different classes of errors.
- _Not_ run the diffing process, since the user very likely meant for the diffing process to do something which includes another flag.

### Current Behavior

`cdk diff --no-such-flag` runs the diff, prints no warnings related to the flag, and in fact does not indicate in any way that there is a problem.

### Reproduction Steps

Run `cdk diff --no-such-flag`.

### CDK CLI Version

2.93.0 (build 724bd01)

### Node.js Version

v18.17.1

### OS

NixOS

### Language

Typescript

### Language Version

5.2.2

**Comment by indrora** [comment]
This I suspect affects much more than the diff command, as we extract the arguments and parameters from the command line through a more blanket mechanism.

**Comment by l0b0** [comment]
Fair enough. It doesn't affect the `cdk` command itself (try `cdk --foo`), so hopefully there is a mechanism already in place which can be used.
